src/Language/PureScript/Make.hs

```diff
@@ -75,9 +76,16 @@ rebuildModule' MakeActions{..} exEnv externs m@(Module _ _ moduleName _ _) = do
   let env = foldl' (flip applyExternsFileToEnvironment) initEnvironment externs
       withPrim = importPrim m
   lint withPrim
+
   ((Module ss coms _ elaborated exps, env'), nextVar) <- runSupplyT 0 $ do
-    desugar exEnv externs [withPrim] >>= \case
-      [desugared] -> runCheck' (emptyCheckState env) $ typeCheckModule desugared
+    runStateT (desugar externs [withPrim]) (exEnv, mempty) >>= \case
+      ([desugared], (exEnv', usedImportsByModuleName)) -> do
+        (checked, CheckState{..}) <- runStateT (typeCheckModule desugared) $ emptyCheckState env
+        let usedImports = fold $ M.lookup moduleName usedImportsByModuleName
+            usedImports' = foldl' (flip $ \(fromModuleName, newtypeCtorName) ->
+              M.alter (Just . (fmap DctorName newtypeCtorName :) . fold) fromModuleName) usedImports checkCoercedNewtypeCtorsImports
+        censor (addHint (ErrorInModule moduleName)) $ lintImports checked exEnv' usedImports'
```

Comment:
Could you please add a comment here saying that we need to wait until after type checking to lint imports because newtype constructors may be marked as used as part of Coercible solving?